### PR TITLE
[MU4] Fix compiler/analyzer warnings found using Xcode

### PR DESCRIPTION
**Pull request by cbjeukendrup**
See the commit messages for details. 

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by cbjeukendrup** [comment]
The code style check fails... but I'm really convinced that the change is correct. Or am I missing something?

File: src/palette/qml/MuseScore/Palette/PaletteTree.qml [hunk 1]
```diff
@@ -29,7 +29,7 @@ import "utils.js" as Utils
 
 ListView {
     id: paletteTree
-    Accessible.name: qsTrc("palette", "Palettes Tree, contains %n palette(s)").arg(count)
+    Accessible.name: qsTrc("palette", "Palettes Tree, contains %1 palette(s)").arg(count)
```
[the remark this answers is not in the document]
Comment (cbjeukendrup) [reply src/palette/qml/MuseScore/Palette/PaletteTree.qml:32]: Ah, that is what it means...!
I thought it was a mistake because there was a warning and I could not find any documentation about `%n`. Now I have finally found [some documentation](https://doc.qt.io/qt-5/i18n-source-translation.html#handling-plurals), so I will have a closer look at it.

File: src/libmscore/read206.cpp [hunk 2]
```diff
@@ -1501,8 +1501,8 @@ bool readNoteProperties206(Note* note, XmlReader& e)
             sp->setTick(e.tick());
             note->addSpannerFor(sp);
             sp->setParent(note);
+            adjustPlacement(sp);
         }
-        adjustPlacement(sp);
```
[the remark this answers is not in the document]
Comment (cbjeukendrup) [reply src/libmscore/read206.cpp:1505]: Yes, that seems to be the case. Probably with many other things too. I am planning to do also a 3.x version, probably next week.

File: src/palette/qml/MuseScore/Palette/PaletteTree.qml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbjeukendrup) [reply src/palette/qml/MuseScore/Palette/PaletteTree.qml]: I found out what was the problem! Thanks Jojo, for making me aware of this!
In 3.x, this problem does not occur, so no need to change anything there.

File: src/libmscore/fraction.h [hunk 3]
```diff
@@ -245,15 +245,15 @@ class Fraction
         if (ticks == -1) {
             return Fraction(-1,1);        // HACK
         }
-        return Fraction(ticks, MScore::division* 4).reduced();
+        return Fraction(ticks, MScore::division * 4).reduced();
```
Comment (Jojo-Schmitz) [inline src/libmscore/fraction.h:248]: Believe it or not: the current code style wants it that way...

File: src/libmscore/fraction.h [hunk 4]
```diff
@@ -245,15 +245,15 @@ class Fraction
...
     }
 
     //---------------------------------------------------------
     //   eps
     ///   A very small fraction, corresponds to 1 MIDI tick
     //---------------------------------------------------------
 
-    static Fraction eps() { return Fraction(1, MScore::division* 4); }
+    static Fraction eps() { return Fraction(1, MScore::division * 4); }
```
Comment (Jojo-Schmitz) [inline src/libmscore/fraction.h:256]: same here

**Comment by igorkorsukov** [comment]
> The code style check fails... but I'm really convinced that the change is correct. Or am I missing something?

Your changes are correct, the code style checker is not working correctly.
So that we can push your changes, you need to change the config of the code style checker so that your code passes the check.
https://github.com/musescore/MuseScore/tree/master/tools/codestyle

File: src/libmscore/segment.cpp [hunk 5]
```diff
@@ -986,7 +986,7 @@ bool Segment::setProperty(Pid propertyId, const QVariant& v)
 qreal Segment::widthInStaff(int staffIdx, SegmentType t) const
 {
     const qreal segX = x();
-    qreal nextSegX = segX;
+    qreal nextSegX;
```
Comment (igorkorsukov) [inline src/libmscore/segment.cpp:989]: Would you tell, why this change was made?

Comment (cbjeukendrup) [reply src/libmscore/segment.cpp:989]: In the lines here below, `nextSegX` will be set to another value anyway, so Xcode gave a warning that the initial value is never read. Therefore, I removed that initial value.

Comment (RomanPudashkin) [reply src/libmscore/segment.cpp:989]: Please, return initial values to all variables with primitive types. Variables with undefined values are very dangerous. After some time implementation of this function may change and this initial value can save from hard reproduced bug

Comment (cbjeukendrup) [reply src/libmscore/segment.cpp]: I see, I returned it!

**Comment by cbjeukendrup** [comment]
@igorkorsukov I'm not sure what I should change in the Uncrustify config. The config seems fine to me, but uncrustify thinks that that star is a pointer star instead of an arithmetic operator. 
If the PR would be merged, would then on all subsequent PRs the code style check fail? Or will uncrustify take it for granted then?
Otherwise, I think this small cosmetic issue is not worth the trouble.

**Comment by Jojo-Schmitz** [comment]
Just adjust your code and let @igorkorsukov amd his team deal with uncrustify

**Comment by cbjeukendrup** [comment]
> Just adjust your code and let @igorkorsukov amd his team deal with uncrustify

That seems indeed the best for now. I reverted the 'controversial' changes, so we can deal with it at some other moment.
